### PR TITLE
r/launch_template - fix placement partition number

### LINKED ISSUE #13236: Invalid aws_launch_template being created with v2.61.0
Sorry I'm a little confused by the bug template, not sure where I'm supposed to explain the bug.

But v2.61.0 is creating invalid launch templates; it creates templates with partition_index 0 and no placement group, which results in this error from AWS when creating an EKS cluster:

```
Error creating AutoScaling Group: ValidationError: You must use a valid fully-formed launch template. You cannot use PartitionNumber with a Placement Group that does not exist. Specify a valid Placement Group and try again.
```

v2.60.0 works fine, the partition number is left empty. Seems to be caused by #11655.

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

Terraform v0.12.24
+ provider.aws v2.61.0
+ provider.local v1.4.0
+ provider.null v2.1.2
+ provider.random v2.2.1
+ provider.template v2.1.2

### Affected Resource(s)

* aws_launch_template

### Terraform Configuration Files

Sorry can't share our actual code, didn't create a minimal example.

### Debug Output

See above.

### Panic Output

None

### Expected Behavior

Can create ASGs based on launch templates.

### Actual Behavior

Creating an ASG based on a launch template created by v2.61.0 causes an error (see top of report).

### Steps to Reproduce

Try to create an ASG with v2.61.0 and no placement group.

### PR DESCRIPTION
**Pull request by DrFaust92**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes #13236
Relates #11655

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
resource_aws_launch_template: fix placement partition number to disallow zero value
```

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccXXX'
--- PASS: TestAccAWSLaunchTemplate_basic (42.44s)
--- PASS: TestAccAWSLaunchTemplate_disappears (28.76s)
--- PASS: TestAccAWSLaunchTemplate_BlockDeviceMappings_EBS (89.40s)
--- PASS: TestAccAWSLaunchTemplate_BlockDeviceMappings_EBS_DeleteOnTermination (113.19s)
--- PASS: TestAccAWSLaunchTemplate_EbsOptimized (149.86s)
--- PASS: TestAccAWSLaunchTemplate_ElasticInferenceAccelerator (67.50s)
--- PASS: TestAccAWSLaunchTemplate_data (41.32s)
--- PASS: TestAccAWSLaunchTemplate_description (68.95s)
--- PASS: TestAccAWSLaunchTemplate_update (121.02s)
--- PASS: TestAccAWSLaunchTemplate_capacityReservation_preference (40.91s)
--- PASS: TestAccAWSLaunchTemplate_capacityReservation_target (51.49s)
--- PASS: TestAccAWSLaunchTemplate_cpuOptions (35.05s)
--- PASS: TestAccAWSLaunchTemplate_creditSpecification_nonBurstable (45.75s)
--- PASS: TestAccAWSLaunchTemplate_creditSpecification_t2 (43.08s)
--- PASS: TestAccAWSLaunchTemplate_creditSpecification_t3 (45.55s)
--- PASS: TestAccAWSLaunchTemplate_IamInstanceProfile_EmptyConfigurationBlock (34.94s)
--- PASS: TestAccAWSLaunchTemplate_networkInterface (85.16s)
--- PASS: TestAccAWSLaunchTemplate_networkInterfaceAddresses (85.63s)
--- PASS: TestAccAWSLaunchTemplate_associatePublicIPAddress (183.46s)
--- PASS: TestAccAWSLaunchTemplate_placement_partitionNum (77.26s)
--- PASS: TestAccAWSLaunchTemplate_networkInterface_ipv6Addresses (41.51s)
--- PASS: TestAccAWSLaunchTemplate_networkInterface_ipv6AddressCount (44.00s)
--- PASS: TestAccAWSLaunchTemplate_instanceMarketOptions (124.94s)
--- PASS: TestAccAWSLaunchTemplate_licenseSpecification (45.93s)
--- PASS: TestAccAWSLaunchTemplate_metadataOptions (41.83s)
--- PASS: TestAccAWSLaunchTemplate_hibernation (97.52s)



--- PASS: TestAccAWSAutoScalingGroup_basic (268.04s)
--- PASS: TestAccAWSAutoScalingGroup_namePrefix (84.81s)
--- PASS: TestAccAWSAutoScalingGroup_autoGeneratedName (83.41s)
--- PASS: TestAccAWSAutoScalingGroup_terminationPolicies (151.82s)
--- PASS: TestAccAWSAutoScalingGroup_tags (278.75s)
--- PASS: TestAccAWSAutoScalingGroup_VpcUpdates (156.81s)
--- PASS: TestAccAWSAutoScalingGroup_WithLoadBalancer (397.34s)
--- PASS: TestAccAWSAutoScalingGroup_WithLoadBalancer_ToTargetGroup (434.09s)
--- PASS: TestAccAWSAutoScalingGroup_withPlacementGroup (184.12s)
--- PASS: TestAccAWSAutoScalingGroup_enablingMetrics (256.30s)
--- PASS: TestAccAWSAutoScalingGroup_suspendingProcesses (309.82s)
--- PASS: TestAccAWSAutoScalingGroup_withMetrics (125.89s)
--- PASS: TestAccAWSAutoScalingGroup_serviceLinkedRoleARN (85.04s)
--- PASS: TestAccAWSAutoScalingGroup_MaxInstanceLifetime (125.63s)
--- PASS: TestAccAWSAutoScalingGroup_ALB_TargetGroups (284.92s)
--- PASS: TestAccAWSAutoScalingGroup_TargetGroupArns (347.68s)
--- PASS: TestAccAWSAutoScalingGroup_initialLifecycleHook (243.39s)
--- PASS: TestAccAWSAutoScalingGroup_ALB_TargetGroups_ELBCapacity (343.04s)
--- PASS: TestAccAWSAutoScalingGroup_classicVpcZoneIdentifier (87.32s)
--- PASS: TestAccAWSAutoScalingGroup_emptyAvailabilityZones (96.53s)
--- PASS: TestAccAWSAutoScalingGroup_launchTemplate (91.43s)
--- PASS: TestAccAWSAutoScalingGroup_launchTemplate_update (217.60s)
--- PASS: TestAccAWSAutoScalingGroup_LaunchTemplate_IAMInstanceProfile (102.81s)
--- PASS: TestAccAWSAutoScalingGroup_LoadBalancers (809.89s)
--- PASS: TestAccAWSAutoScalingGroup_MixedInstancesPolicy (88.17s)
--- PASS: TestAccAWSAutoScalingGroup_MixedInstancesPolicy_InstancesDistribution_OnDemandAllocationStrategy (87.23s)
--- PASS: TestAccAWSAutoScalingGroup_MixedInstancesPolicy_InstancesDistribution_OnDemandBaseCapacity (150.81s)
--- PASS: TestAccAWSAutoScalingGroup_MixedInstancesPolicy_InstancesDistribution_OnDemandPercentageAboveBaseCapacity (128.11s)
--- PASS: TestAccAWSAutoScalingGroup_MixedInstancesPolicy_InstancesDistribution_SpotAllocationStrategy (101.97s)
--- PASS: TestAccAWSAutoScalingGroup_MixedInstancesPolicy_InstancesDistribution_SpotInstancePools (128.51s)
--- PASS: TestAccAWSAutoScalingGroup_MixedInstancesPolicy_InstancesDistribution_SpotMaxPrice (151.75s)
--- PASS: TestAccAWSAutoScalingGroup_MixedInstancesPolicy_LaunchTemplate_LaunchTemplateSpecification_LaunchTemplateName (89.97s)
--- PASS: TestAccAWSAutoScalingGroup_MixedInstancesPolicy_LaunchTemplate_LaunchTemplateSpecification_Version (127.05s)
--- PASS: TestAccAWSAutoScalingGroup_MixedInstancesPolicy_LaunchTemplate_Override_InstanceType (142.73s)
--- PASS: TestAccAWSAutoScalingGroup_MixedInstancesPolicy_LaunchTemplate_Override_WeightedCapacity (177.83s)
--- PASS: TestAccAWSAutoScalingGroup_launchTempPartitionNum (217.61s)

```

### CODE REVIEW

**Comment by ewbankkit** [comment]
Verified acceptance tests:

```console
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSLaunchTemplate_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws/ -v -count 1 -parallel 20 -run=TestAccAWSLaunchTemplate_ -timeout 120m
=== RUN   TestAccAWSLaunchTemplate_basic
=== PAUSE TestAccAWSLaunchTemplate_basic
=== RUN   TestAccAWSLaunchTemplate_disappears
=== PAUSE TestAccAWSLaunchTemplate_disappears
=== RUN   TestAccAWSLaunchTemplate_BlockDeviceMappings_EBS
=== PAUSE TestAccAWSLaunchTemplate_BlockDeviceMappings_EBS
=== RUN   TestAccAWSLaunchTemplate_BlockDeviceMappings_EBS_DeleteOnTermination
=== PAUSE TestAccAWSLaunchTemplate_BlockDeviceMappings_EBS_DeleteOnTermination
=== RUN   TestAccAWSLaunchTemplate_EbsOptimized
=== PAUSE TestAccAWSLaunchTemplate_EbsOptimized
=== RUN   TestAccAWSLaunchTemplate_ElasticInferenceAccelerator
=== PAUSE TestAccAWSLaunchTemplate_ElasticInferenceAccelerator
=== RUN   TestAccAWSLaunchTemplate_data
=== PAUSE TestAccAWSLaunchTemplate_data
=== RUN   TestAccAWSLaunchTemplate_description
=== PAUSE TestAccAWSLaunchTemplate_description
=== RUN   TestAccAWSLaunchTemplate_update
=== PAUSE TestAccAWSLaunchTemplate_update
=== RUN   TestAccAWSLaunchTemplate_tags
=== PAUSE TestAccAWSLaunchTemplate_tags
=== RUN   TestAccAWSLaunchTemplate_capacityReservation_preference
=== PAUSE TestAccAWSLaunchTemplate_capacityReservation_preference
=== RUN   TestAccAWSLaunchTemplate_capacityReservation_target
=== PAUSE TestAccAWSLaunchTemplate_capacityReservation_target
=== RUN   TestAccAWSLaunchTemplate_cpuOptions
=== PAUSE TestAccAWSLaunchTemplate_cpuOptions
=== RUN   TestAccAWSLaunchTemplate_creditSpecification_nonBurstable
=== PAUSE TestAccAWSLaunchTemplate_creditSpecification_nonBurstable
=== RUN   TestAccAWSLaunchTemplate_creditSpecification_t2
=== PAUSE TestAccAWSLaunchTemplate_creditSpecification_t2
=== RUN   TestAccAWSLaunchTemplate_creditSpecification_t3
=== PAUSE TestAccAWSLaunchTemplate_creditSpecification_t3
=== RUN   TestAccAWSLaunchTemplate_IamInstanceProfile_EmptyConfigurationBlock
=== PAUSE TestAccAWSLaunchTemplate_IamInstanceProfile_EmptyConfigurationBlock
=== RUN   TestAccAWSLaunchTemplate_networkInterface
=== PAUSE TestAccAWSLaunchTemplate_networkInterface
=== RUN   TestAccAWSLaunchTemplate_networkInterfaceAddresses
=== PAUSE TestAccAWSLaunchTemplate_networkInterfaceAddresses
=== RUN   TestAccAWSLaunchTemplate_associatePublicIPAddress
=== PAUSE TestAccAWSLaunchTemplate_associatePublicIPAddress
=== RUN   TestAccAWSLaunchTemplate_placement_partitionNum
=== PAUSE TestAccAWSLaunchTemplate_placement_partitionNum
=== RUN   TestAccAWSLaunchTemplate_networkInterface_ipv6Addresses
=== PAUSE TestAccAWSLaunchTemplate_networkInterface_ipv6Addresses
=== RUN   TestAccAWSLaunchTemplate_networkInterface_ipv6AddressCount
=== PAUSE TestAccAWSLaunchTemplate_networkInterface_ipv6AddressCount
=== RUN   TestAccAWSLaunchTemplate_instanceMarketOptions
=== PAUSE TestAccAWSLaunchTemplate_instanceMarketOptions
=== RUN   TestAccAWSLaunchTemplate_licenseSpecification
=== PAUSE TestAccAWSLaunchTemplate_licenseSpecification
=== RUN   TestAccAWSLaunchTemplate_metadataOptions
=== PAUSE TestAccAWSLaunchTemplate_metadataOptions
=== RUN   TestAccAWSLaunchTemplate_hibernation
=== PAUSE TestAccAWSLaunchTemplate_hibernation
=== CONT  TestAccAWSLaunchTemplate_basic
=== CONT  TestAccAWSLaunchTemplate_hibernation
=== CONT  TestAccAWSLaunchTemplate_metadataOptions
=== CONT  TestAccAWSLaunchTemplate_licenseSpecification
=== CONT  TestAccAWSLaunchTemplate_instanceMarketOptions
=== CONT  TestAccAWSLaunchTemplate_networkInterface_ipv6AddressCount
=== CONT  TestAccAWSLaunchTemplate_networkInterface_ipv6Addresses
=== CONT  TestAccAWSLaunchTemplate_placement_partitionNum
=== CONT  TestAccAWSLaunchTemplate_associatePublicIPAddress
=== CONT  TestAccAWSLaunchTemplate_networkInterfaceAddresses
=== CONT  TestAccAWSLaunchTemplate_networkInterface
=== CONT  TestAccAWSLaunchTemplate_IamInstanceProfile_EmptyConfigurationBlock
=== CONT  TestAccAWSLaunchTemplate_creditSpecification_t3
=== CONT  TestAccAWSLaunchTemplate_creditSpecification_t2
=== CONT  TestAccAWSLaunchTemplate_creditSpecification_nonBurstable
=== CONT  TestAccAWSLaunchTemplate_cpuOptions
=== CONT  TestAccAWSLaunchTemplate_capacityReservation_target
=== CONT  TestAccAWSLaunchTemplate_capacityReservation_preference
=== CONT  TestAccAWSLaunchTemplate_tags
=== CONT  TestAccAWSLaunchTemplate_update
--- PASS: TestAccAWSLaunchTemplate_IamInstanceProfile_EmptyConfigurationBlock (25.94s)
=== CONT  TestAccAWSLaunchTemplate_description
--- PASS: TestAccAWSLaunchTemplate_cpuOptions (27.35s)
=== CONT  TestAccAWSLaunchTemplate_data
--- PASS: TestAccAWSLaunchTemplate_creditSpecification_nonBurstable (29.00s)
=== CONT  TestAccAWSLaunchTemplate_ElasticInferenceAccelerator
--- PASS: TestAccAWSLaunchTemplate_networkInterface_ipv6AddressCount (29.34s)
=== CONT  TestAccAWSLaunchTemplate_EbsOptimized
--- PASS: TestAccAWSLaunchTemplate_creditSpecification_t2 (29.72s)
=== CONT  TestAccAWSLaunchTemplate_BlockDeviceMappings_EBS_DeleteOnTermination
--- PASS: TestAccAWSLaunchTemplate_networkInterface_ipv6Addresses (29.72s)
=== CONT  TestAccAWSLaunchTemplate_BlockDeviceMappings_EBS
--- PASS: TestAccAWSLaunchTemplate_basic (29.95s)
=== CONT  TestAccAWSLaunchTemplate_disappears
--- PASS: TestAccAWSLaunchTemplate_metadataOptions (30.32s)
--- PASS: TestAccAWSLaunchTemplate_creditSpecification_t3 (30.38s)
--- PASS: TestAccAWSLaunchTemplate_capacityReservation_preference (30.76s)
--- PASS: TestAccAWSLaunchTemplate_licenseSpecification (31.60s)
--- PASS: TestAccAWSLaunchTemplate_capacityReservation_target (35.71s)
--- PASS: TestAccAWSLaunchTemplate_tags (48.79s)
--- PASS: TestAccAWSLaunchTemplate_disappears (18.86s)
--- PASS: TestAccAWSLaunchTemplate_networkInterfaceAddresses (49.64s)
--- PASS: TestAccAWSLaunchTemplate_networkInterface (49.70s)
--- PASS: TestAccAWSLaunchTemplate_placement_partitionNum (50.44s)
--- PASS: TestAccAWSLaunchTemplate_data (25.38s)
--- PASS: TestAccAWSLaunchTemplate_hibernation (63.85s)
--- PASS: TestAccAWSLaunchTemplate_description (41.48s)
--- PASS: TestAccAWSLaunchTemplate_ElasticInferenceAccelerator (41.41s)
--- PASS: TestAccAWSLaunchTemplate_BlockDeviceMappings_EBS (51.49s)
--- PASS: TestAccAWSLaunchTemplate_update (83.01s)
--- PASS: TestAccAWSLaunchTemplate_instanceMarketOptions (84.73s)
--- PASS: TestAccAWSLaunchTemplate_associatePublicIPAddress (100.19s)
--- PASS: TestAccAWSLaunchTemplate_BlockDeviceMappings_EBS_DeleteOnTermination (80.39s)
--- PASS: TestAccAWSLaunchTemplate_EbsOptimized (90.91s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	120.313s
```

File: aws/resource_aws_autoscaling_group_test.go [hunk 1]
```diff
@@ -4105,3 +4140,57 @@ resource "aws_autoscaling_group" "test" {
...
+  name     = %[1]q
+  strategy = "cluster"
+}
+
+resource "aws_autoscaling_group" "test" {
+  name_prefix        = "test"
+  availability_zones = ["${data.aws_availability_zones.available.names[0]}"]
+  desired_capacity   = 1
+  min_size           = 1
+  max_size           = 3
```
Comment (bflad) [inline aws/resource_aws_autoscaling_group_test.go:4188]: We don't need to actually spin up instances for the Launch Template errors to occur. 👍 Verified locally.

```suggestion
  min_size           = 0
  max_size           = 0
```

File: aws/resource_aws_autoscaling_group_test.go [hunk 2]
```diff
@@ -2069,6 +2069,41 @@ func TestAccAWSAutoScalingGroup_MixedInstancesPolicy_LaunchTemplate_Override_Wei
...
+func TestAccAWSAutoScalingGroup_launchTempPartitionNum(t *testing.T) {
+	var group autoscaling.Group
+
+	randName := fmt.Sprintf("terraform-test-%s", acctest.RandString(10))
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:            func() { testAccPreCheck(t) },
+		Providers:           testAccProviders,
+		CheckDestroy:        testAccCheckAWSAutoScalingGroupDestroy,
+		DisableBinaryDriver: true,
```
Comment (bflad) [inline aws/resource_aws_autoscaling_group_test.go:2081]: Nit: This temporary flag is only necessary for acceptance testing that is checking `TypeSet` values at the moment.

**Comment by bflad** [comment]
Fun enough, it looks like the EC2 API _just_ was updated to be able to return validation errors (thank goodness): https://github.com/terraform-providers/terraform-provider-aws/pull/13190/files#diff-6fed94e5d96961410c0aedf93f473c1fR46604 -- I will create a separate issue for that.

**Pull request merged**